File: api/ExpressedRealms.Server/EndPoints/CharacterEndPoints/Responses/CharacterSkillsResponse.cs
namespace ExpressedRealms.Server.EndPoints.CharacterEndPoints.Responses;

public class CharacterSkillsResponse
{
    public byte SkillTypeId { get; set; }
    public string Name { get; set; } = null!;
    public string Description { get; set; } = null!;
    public byte LevelId { get; set; }
    public string LevelName { get; set; } = null!;
    public string LevelDescription { get; set; } = null!;
    public byte SkillSubTypeId { get; set; }
    public List<BenefitItemResponse> Benefits { get; set; } = new();
    public int XP { get; set; }
}

File: client/src/components/characters/character/skills/interfaces/CharacterSkillsResponse.ts
import type {BenefitItemResponse} from "@/components/characters/character/interfaces/BenefitItemResponse";

export interface CharacterSkillsResponse {
    skillTypeId: number;
    name: string;
    description: string;
    levelId: number;
    levelName: string;
    levelDescription: string;
    skillSubTypeId: number;
    xp: number;
    levelNumber: number;
}
